### PR TITLE
make update-or-insert infallible op

### PR DESCRIPTION
**Pull request by taozhu-chicago**
#### Problem

`upsert` (update-or-insert) should be infallible, querying result should be lazy;

#### Summary of Changes

- Made `upsert` infallible

Fixes #

### CODE REVIEW

File: runtime/src/cost_model.rs [hunk 1]
```diff
@@ -116,17 +100,11 @@ impl CostModel {
         tx_cost
     }
 
-    pub fn upsert_instruction_cost(
-        &mut self,
-        program_key: &Pubkey,
-        cost: u64,
-    ) -> Result<u64, &'static str> {
+    // update-or-insert should be infallible. Query result of upsert, which often
+    // requires additional calculation, should be lazy.
```
Comment (carllin) [inline runtime/src/cost_model.rs:104]: nit: triple `///` For function documentation

This comment is confusing as it doesn't describe what the function is doing or provide any insight into how the function should be used. It seems to describe some design decision, but isn't very clear.

**Review by t-nelson — APPROVED** [review]

**Review by carllin — APPROVED** [review]
Thanks for the cleanup! This is just removing the `dirty` bool from the `update_cost_model()` function right? 

In the future, it would be helpful if the `Problem` statement in the PR was more explicit about describing the impetus for the change.

For instance maybe "update_cost_model() return value is unnecessary".

File: runtime/src/cost_model.rs (hunk 1 above)
Comment (t-nelson) [reply runtime/src/cost_model.rs:104]: It's more easily reviewed from https://github.com/solana-labs/solana/pull/23651/files?w=1

Comment (taozhu-chicago) [reply runtime/src/cost_model.rs:104]: right, good point. Will remove this comment next time near it

**Pull request merged**